src/main/java/com/example/Restaurant_Management_System_REST_API/service/ReportService.java

```diff
@@ -0,0 +1,392 @@
+        Table table = tableRepository.findById(id).orElseThrow(() -> new NotFoundInDatabaseException(Table.class));
+
+        return modelMapper.map(table, TableForReportDTO.class);
+    }
+
+    public List<Table> getTableByAvailability(boolean isAvailable) {
+        return tableRepository.findByIsAvailable(isAvailable);
+    }
+
+    public double getTotalSumRestaurantOrdersInPeriodTime(LocalDate timeFrom, LocalDate timeTo) {
+        double totalSpent = 0;
+
+        List<RestaurantOrder> allRestaurantOrdersInSpecificTimePeriod = restaurantOrderRepository.findByTimeRange(timeFrom.atStartOfDay(),
+                timeTo.plusDays(1).atStartOfDay());
+
+        for (RestaurantOrder restaurantOrder : allRestaurantOrdersInSpecificTimePeriod) {
+            double totalAmountToPayPerRestaurantOrder = restaurantOrder.getTotalAmountToPay();
+            totalSpent += totalAmountToPayPerRestaurantOrder;
+        }
+        return totalSpent;
+    }
+
+    public RestaurantOrderMenuRecordDTO getMenuRecordsFromRestaurantOrderId(Long id) throws NotFoundInDatabaseException {
+        List<MenuRecordForOrderDTO> menuRecordForOrderDTOList = new ArrayList<>();
+
+        //I use here object RestaurantOrder taken straight from repo because when I use service I got object without portions amount
+        RestaurantOrder restaurantOrder = restaurantOrderRepository.findById(id).orElseThrow(() -> new NotFoundInDatabaseException(RestaurantOrder.class));
+
+        // I fetch necessary data from repo. this is the list of RestaurantOrderMenuRecord given for specific RestaurantOrder
+        List<RestaurantOrderMenuRecord> restaurantOrderMenuRecordList = restaurantOrderMenuRecordRepository.findRestaurantOrderMenuRecordByRestaurantOrderId(id);
+
+        //Now I need to loop over this fetched list. I take menuRecords from this list. Then I convert it to menuRecordForOrderDTO
+        // and I set the portions amount and finally add it to the list of MenuRecordForOrderDTO
+        for (RestaurantOrderMenuRecord eachRestaurantOrderMenuRecord : restaurantOrderMenuRecordList) {
+            MenuRecord menuRecord = eachRestaurantOrderMenuRecord.getMenuRecord();
+            MenuRecordForOrderDTO menuRecordForOrderDTO = modelMapper.map(menuRecord, MenuRecordForOrderDTO.class);
+            menuRecordForOrderDTO.setPortionsAmount(eachRestaurantOrderMenuRecord.getPortionsAmount());
+            menuRecordForOrderDTOList.add(menuRecordForOrderDTO);
+        }
+
+        RestaurantOrderMenuRecordDTO restaurantOrderMenuRecordDTO = new RestaurantOrderMenuRecordDTO();
+        restaurantOrderMenuRecordDTO.setMenuRecord(menuRecordForOrderDTOList);
+        restaurantOrderMenuRecordDTO.setTotalAmountToPay(restaurantOrder.getTotalAmountToPay());
+
+
+        return restaurantOrderMenuRecordDTO;
+    }
+
+    public StreamingResponseBody getRestaurantOrderMenuRecordInTimePeriod(LocalDate timeFrom, LocalDate timeTo) {
```

Comment:
Variable names here are misleading. Is it date or is it time?

Reply:
date